App crashes when permissions are denied and the app is resumed after minimizing in Redmi 4A

**Actual Behaviour**
App crashes when permissions are denied and the app is resumed after minimizing in Redmi 4A. When partial or no permissions are given and app is minimized, on resuming a runtime null pointer exception occurs. Screenshot attached.

**Expected Behaviour**
No crash should happen.

**Steps to reproduce it**
Give partial or no permission in Android N on Redmi 4A (MiUI 10)

**LogCat for the issue**
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.google.android.packageinstaller/com.android.packageinstaller.permission.ui.GrantPermissionsActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2723)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2784)
	at android.app.ActivityThread.-wrap12(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1523)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:163)
	at android.app.ActivityThread.main(ActivityThread.java:6238)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:904)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:794)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
	at com.android.packageinstaller.permission.ui.GrantPermissionsActivity.computePermissionGrantState(GrantPermissionsActivity.java:318)
	at com.android.packageinstaller.permission.ui.GrantPermissionsActivity.updateDefaultResults(GrantPermissionsActivity.java:367)
	at com.android.packageinstaller.permission.ui.GrantPermissionsActivity.onCreate(GrantPermissionsActivity.java:124)
	at android.app.Activity.performCreate(Activity.java:6868)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2676)
	... 9 more

**Screenshots of the issue**
![whatsapp image 2018-12-10 at 10 37 47 pm 1](https://user-images.githubusercontent.com/34205476/49749022-3a375480-fccd-11e8-8710-4e2b92a41f00.jpeg)
![whatsapp image 2018-12-10 at 10 37 47 pm](https://user-images.githubusercontent.com/34205476/49749033-402d3580-fccd-11e8-8561-98710225c680.jpeg)


**Would you like to work on the issue?**
No

--- a/app/src/main/java/org/fossasia/susi/ai/chat/ChatActivity.kt
+++ b/app/src/main/java/org/fossasia/susi/ai/chat/ChatActivity.kt
@@ -389,26 +389,25 @@ class ChatActivity : AppCompatActivity(), IChatView {
         ActivityCompat.requestPermissions(this, permissions, PERM_REQ_CODE)
     }
 
-    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
+    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String?>, grantResults: IntArray) {
         when (requestCode) {
             PERM_REQ_CODE -> run {
                 var audioPermissionGiven = false
-                for (i in permissions.indices) {
-                    when (permissions[i]) {
-                        Manifest.permission.ACCESS_FINE_LOCATION -> if (grantResults.isNotEmpty() && grantResults[i] == PackageManager.PERMISSION_GRANTED) {
+                for ((index, permission) in permissions.withIndex()) {
+                    if (permission == null)
+                        continue
+                    val granted = grantResults.isNotEmpty() && grantResults[index] == PackageManager.PERMISSION_GRANTED
+                    when (permission) {
+                        Manifest.permission.ACCESS_FINE_LOCATION -> if (granted) {
                             chatPresenter.getLocationFromLocationService()
                         }
 
                         Manifest.permission.RECORD_AUDIO -> {
-                            if (grantResults.isNotEmpty() && grantResults[i] == PackageManager.PERMISSION_GRANTED) {
-                                chatPresenter.disableMicInput(false)
-                            } else {
-                                chatPresenter.disableMicInput(true)
-                            }
-                            audioPermissionGiven = true
+                            chatPresenter.disableMicInput(!granted)
+                            audioPermissionGiven = granted
                         }
 
-                        Manifest.permission.WRITE_EXTERNAL_STORAGE -> if (grantResults.size >= 0 && grantResults[i] == PackageManager.PERMISSION_GRANTED && audioPermissionGiven) {
+                        Manifest.permission.WRITE_EXTERNAL_STORAGE -> if (granted && audioPermissionGiven) {
                             chatPresenter.initiateHotwordDetection()
                         }
                     }

